Add improved GUI

LegitProgrammer opened the pull request:
closes #30, closes #31, closes #35

Implemented a new version of GUI that has a consistent and improved look, and integrates all of the features that the application currently offers. 


### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

platisd committed:
Add dummy file [DO NOT MERGE]

Peym77 committed:
#33 #34 Provide API for monitoring state changes when executing commands

Fixed a small bug with getPing functionality

Peym77 committed:
#33 #34 Add vehicle state monitoring

LegitProgrammer committed:
Implement improved gui

* #30, #35 Add user login screen

* #30 Add settings screen

* #30 Add text from .txt file loading functionality

* #30 Add .txt file loading functionality

* #30, #6 Add button for microphone input

* #30, #5 Add .mp3 file loading functionality

* #30 Connect manual and autonomous steering to the new gui

* #31 Add escape hatch to all screens

* #35 Add user MQTT credential saving to file functionality

* #35 Add user MQTT credential loading from file functionality

helps #5, helps #6, resolves #30, resolves #31, resolves #35

Peym77 committed:
#33 #34 Provide more accurate and precise command execution

LegitProgrammer committed:
Merge pull request #44 from DIT113-V22/test_dod_action

Add DOD

LegitProgrammer committed:
Merge pull request #45 from DIT113-V22/car-api-update

Add Car API update

LegitProgrammer committed:
Delete dummy_file

Deleting the dummy file that was accidentally merged.

LegitProgrammer committed:
Add missing css file

Added a css file that was not pushed by mistake during the previous commit.

LegitProgrammer committed:
Merge remote-tracking branch 'origin/gui' into gui

Peym77 reviewed (dismissed):
GUI was manually tested and everything is working as expected. Approved.

LegitProgrammer committed:
#35 Fix a bug with profile creation

Fixed a bug that prevented some profiles from being created.

Peym77 reviewed (changes requested):
Textbox description for the "Connection Address" should be changed to "Port".

LegitProgrammer committed:
#35 Rename connection address to port

Renamed connection address to port in order to make it accurate with it's intended function.

Peym77 reviewed (approved):
Approved.
